src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -1050,10 +1055,41 @@ public MappedByteBuffer map(MapMode mode, long position, long size) throws IOExc
         }
     }
 
-    public Unmapper mapInternal(MapMode mode, long position, long size) throws IOException {
-        boolean isSync = isSync(Objects.requireNonNull(mode, "Mode is null"));
+    private static final int MAP_MEM_SEG_DEFAULT_MODES = 0;
+    private static final int MAP_MEM_SEG_READ_ONLY = 1;
+
+    @Override
+    public MemorySegment map(MapMode mode, long offset, long size,
+                             MemorySession session)
+            throws IOException, UnsupportedOperationException
+    {
+        Objects.requireNonNull(mode,"Mode is null");
+        Objects.requireNonNull(session, "Session is null");
+        MemorySessionImpl sessionImpl = MemorySessionImpl.toSessionImpl(session);
+        sessionImpl.checkValidStateSlow();
+        if (offset < 0) throw new IllegalArgumentException("Requested bytes offset must be >= 0.");
+        if (size < 0) throw new IllegalArgumentException("Requested bytes size must be >= 0.");
```

Comment:
The javadoc also says that IAE will be thrown if `offset + size < 0` I think to guard against overflow, but I don't see that checked here. Is it missing?

Reply:
`mapInternal` in FileChannelImpl takes care of that for both flavors of `map`